tests/fixtures/valid/shared-value-coercion-conditional-explicit.txt

```diff
@@ -0,0 +1,9 @@
+import {useSharedValue, useDerivedValue} from 'react-native-reanimated';
+
+const Component: FC = () => {
+  const foo = useSharedValue(0);
+
+  const bar = useDerivedValue(() => !!foo ? 1 : 0);
```

Comment:
I figured that an explicit `!!foo` (or correspondingly `foo.toString()` for string coercion) should be allowed to "insist" on treating the shared value as a boolean?

But I could see the argument that `!!foo` should also still get flagged